osgeo_importer/validators.py

```diff
@@ -0,0 +1,58 @@
+import os
+from .utils import NoDataSourceFound, load_handler
+from .importers import OSGEO_IMPORTER, VALID_EXTENSIONS
+import logging
+
+logger = logging.getLogger(__name__)
+
+NONDATA_EXTENSIONS = ['shx', 'prj', 'dbf', 'xml', 'sld']
+
+
+def validate_extension(filename):
+    filedir, file = os.path.split(filename)
+    base, extension = os.path.splitext(file)
+    extension = extension.lstrip('.').lower()
+    if extension not in VALID_EXTENSIONS:
+        return False
+    else:
+        return True
+
+
+def validate_shapefiles_have_all_parts(filenamelist):
+    shp = []
+    prj = []
+    dbf = []
+    shx = []
+    for file in filenamelist:
+        base, extension = os.path.splitext(file)
+        extension = extension.lstrip('.').lower()
+        if extension == 'shp':
+            shp.append(base)
+        elif extension == 'prj':
+            prj.append(base)
+        elif extension == 'dbf':
+            dbf.append(base)
+        elif extension == 'shx':
+            shx.append(base)
+    if set(shp) == set(prj) == set(dbf) == set(shx):
+        return True
+    else:
+        return False
+
+
+def validate_inspector_can_read(filename):
```

Comment:
I'm not sure I follow your intent on this @bitner, and I'm thinking this is related to the zip file issue -- why are we shadowing `models.validate_inspector_can_read()` here?